pims/base_frames.py

```diff
@@ -528,3 +528,268 @@ def process(img_or_iterable, *args, **kwargs):
                        "other objects, its behavior is "
                        "unchanged.\n\n") + process.__doc__
     return process
+
+
+class FramesDimension(object):
```

Comment:
The other thing that occurs to me is we should be aware of https://github.com/xray/xray and see what we can steal from them.